### PR TITLE
Add deleting confirmation step to Longhorn uninstallation

### PR DESCRIPTION
**Pull request by PhanLe1010**
longhorn/longhorn#4509

### CODE REVIEW

**Review by shuo-wu — DISMISSED** [review]
May need to update default-setting.yaml as well.
In general LGTM

**Comment by PhanLe1010** [comment]
> May need to update default-setting.yaml as well.

File: types/setting.go [hunk 1]
```diff
@@ -843,6 +846,18 @@ var (
 		ReadOnly: false,
 		Default:  "1",
 	}
+
+	SettingDefinitionDeletingConfirmationFlag = SettingDefinition{
+		DisplayName: "Deleting Confirmation Flag",
```
Comment (innobead) [inline types/setting.go:851]: `Delete Confirmation Flag` instead?

File: types/setting.go [hunk 2]
```diff
@@ -920,6 +935,10 @@ func ValidateSetting(name, value string) (err error) {
 		if value != "true" && value != "false" {
 			return fmt.Errorf("value %v of setting %v should be true or false", value, sName)
 		}
+	case SettingNameDeletingConfirmationFlag:
+		if value != "true" && value != "false" {
+			return fmt.Errorf("value %v of setting %v should be true or false", value, sName)
+		}
```
Comment (innobead) [inline types/setting.go:941]: The implementation is the same as other bool setting, so why not just add this case before `SettingNameUpgradeChecker` then `fallthrough`?

File: controller/uninstall_controller.go [hunk 3]
```diff
@@ -325,6 +325,14 @@ func (c *UninstallController) uninstall() error {
 }
 
 func (c *UninstallController) checkPreconditions() error {
+	confirmationFlag, err := c.ds.GetSettingAsBool(types.SettingNameDeletingConfirmationFlag)
+	if err != nil {
+		return errors.Wrapf(err, "failed to check deleting-confirmation-flag setting")
+	}
+	if !confirmationFlag {
+		return fmt.Errorf("cannot uninstall Longhorn because deleting-confirmation-flag is set to `false`. Please set it to `true` using Longhorn UI or kubectl -n longhorn-system edit settings.longhorn.io deleting-confirmation-flag ")
```
Comment (innobead) [inline controller/uninstall_controller.go:333]: The message is quite long. Could we break with a new line for better reading?

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, some minor requests are required to resolve.

File: controller/uninstall_controller.go (hunk 3 above)
Comment (PhanLe1010) [reply controller/uninstall_controller.go]: Thank you. Fixed

File: types/setting.go (hunk 2 above)
Comment (PhanLe1010) [reply types/setting.go]: Ah. I see. Thank you. Fixed

**Review by innobead — APPROVED** [review]

**Pull request merged**